framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -366,34 +366,42 @@ void MediaPlayerImpl::pausePlayer()
 	mCurState = PLAYER_STATE_PAUSED;
 }
 
-int MediaPlayerImpl::getVolume()
+player_result_t MediaPlayerImpl::getVolume(uint8_t *vol)
 {
-	int ret = -1;
+	player_result_t ret = PLAYER_OK;
 
 	std::unique_lock<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaPlayer getVolume\n");
 
+	if (vol == nullptr) {
+		meddbg("The given argument is invalid.\n");
+		return PLAYER_ERROR_INVALID_PARAM;
+	}
+
 	PlayerWorker& mpw = PlayerWorker::getWorker();
 	if (!mpw.isAlive()) {
 		meddbg("PlayerWorker is not alive\n");
-		return -1;
+		return PLAYER_ERROR;
 	}
 
-	mpw.enQueue(&MediaPlayerImpl::getVolumePlayer, shared_from_this(), std::ref(ret));
+	mpw.enQueue(&MediaPlayerImpl::getVolumePlayer, shared_from_this(), vol, std::ref(ret));
 	mSyncCv.wait(lock);
 
 	return ret;
 }
 
-void MediaPlayerImpl::getVolumePlayer(int &ret)
+void MediaPlayerImpl::getVolumePlayer(uint8_t *vol, player_result_t &ret)
 {
 	medvdbg("MediaPlayer Worker : getVolume\n");
+	if (get_output_audio_volume(vol) != AUDIO_MANAGER_SUCCESS) {
+		meddbg("get_output_audio_volume() is failed, ret = %d\n", ret);
```

Comment:
PLAYER_ERROR_INVALID_OPERATION is aligned with AUDIO_MANAGER_DEVICE_NOT_SUPPORT.
You decided policy that return default volume if it is excluded, PLAYER_ERROR_INVALID_OPERATION is not acceptable for getvolume

Reply:
Right. I just changed as 'ret = PLAYER_ERROR'.